dedalus/core/basis.py

```diff
@@ -4969,4 +5051,83 @@ def operate(self, out):
         np.copyto(out.data, operand.data[axslice(self.index,0,2)])
 
 
+class CartesianAdvectiveCFL(operators.AdvectiveCFL):
+
+    input_coord_type = CartesianCoordinates
+    input_basis_type = (ComplexFourier, RealFourier, Jacobi)
+
+    @CachedMethod
+    def grid_spacing(self, velocity):
+        coordsys = velocity.tensorsig[0]
+        grid_spacing = np.zeros_like(velocity.data)
```

Comment:
`grid_spacing` should be a list of broadcastable 1D vectors -- should not need to be the full shape of `velocity.data`

Reply:
Done, I think. This un-vectorizes the division by spacing in compute_cfl_frequency() here: https://github.com/evanhanders/dedalus/blob/d3_cfl/dedalus/core/basis.py#L5147 (and makes it a for-loop).